src/ring_theory/witt_vector/structure_polynomial.lean

```diff
@@ -129,9 +133,132 @@ begin
     funext k, exact H k },
 end
 
+lemma witt_structure_rat_rec_aux (Φ : mv_polynomial idx ℚ) (n : ℕ) :
+  (witt_structure_rat p Φ n) * C (p ^ n : ℚ) =
+  ((bind₁ (λ b, (rename (λ i, (b, i)) (W_ ℚ n))) Φ)) -
+  ∑ i in range n, C (p ^ i : ℚ) * (witt_structure_rat p Φ i) ^ p ^ (n - i) :=
+begin
+  have := X_in_terms_of_W_aux p ℚ n,
+  replace := congr_arg (bind₁ (λ k : ℕ, (bind₁ (λ i, (rename (prod.mk i) (W_ ℚ k)))) Φ)) this,
+  rw [alg_hom.map_mul, bind₁_C_right] at this,
+  convert this, clear this,
+  conv_rhs { simp only [alg_hom.map_sub, bind₁_X_right] },
+  rw sub_right_inj,
+  simp only [alg_hom.map_sum, alg_hom.map_mul, bind₁_C_right, alg_hom.map_pow],
+  refl
+end
+
+lemma witt_structure_rat_rec (Φ : mv_polynomial idx ℚ) (n : ℕ) :
+  (witt_structure_rat p Φ n) = C (1 / p ^ n : ℚ) *
+  (bind₁ (λ b, (rename (λ i, (b, i)) (W_ ℚ n))) Φ -
+  ∑ i in range n, C (p ^ i : ℚ) * (witt_structure_rat p Φ i) ^ p ^ (n - i)) :=
+begin
+  rw [← witt_structure_rat_rec_aux p Φ n, mul_comm, mul_assoc,
+      ← C_mul, mul_one_div_cancel, C_1, mul_one],
+  exact pow_ne_zero _ (nat.cast_ne_zero.2 $ ne_of_gt (nat.prime.pos ‹_›)),
+end
+
+/-- `witt_structure_int Φ` is a family of polynomials `ℕ → mv_polynomial (idx × ℕ) ℚ`
+that are uniquely characterised by the property that
+`bind₁ (witt_structure_int p Φ) (witt_polynomial p ℚ n) = bind₁ (λ i, (rename (prod.mk i) (witt_polynomial p ℚ n))) Φ`.
+In other words: evaluating the `n`-th Witt polynomial on the family `witt_structure_int Φ`
+is the same as evaluation `Φ` on the (appropriately renamed) `n`-th Witt polynomials.
+
+See `witt_structure_int_prop` for this property,
+and `witt_structure_int_exists_unique` for the fact that `witt_structure_int`
+gives the unique family of polynomials with this property. -/
+noncomputable def witt_structure_int (Φ : mv_polynomial idx ℤ) (n : ℕ) : mv_polynomial (idx × ℕ) ℤ :=
+finsupp.map_range rat.num (rat.coe_int_num 0)
+  (witt_structure_rat p (map (int.cast_ring_hom ℚ) Φ) n)
+
+variable {p}
+
+lemma bind₁_rename_expand_witt_polynomial (Φ : mv_polynomial idx ℤ) (n : ℕ)
+  (IH : ∀ m : ℕ, m < (n + 1) →
+    map (int.cast_ring_hom ℚ) (witt_structure_int p Φ m) = witt_structure_rat p (map (int.cast_ring_hom ℚ) Φ) m) :
+  bind₁ (λ b, rename (λ i, (b, i)) (expand p (W_ ℤ n))) Φ =
+  bind₁ (λ i, expand p (witt_structure_int p Φ i)) (W_ ℤ n) :=
+begin
+  apply mv_polynomial.map_injective (int.cast_ring_hom ℚ) int.cast_injective,
+  simp only [map_bind₁, map_rename, map_expand, rename_expand, map_witt_polynomial],
+  have key := (witt_structure_rat_prop p (map (int.cast_ring_hom ℚ) Φ) n).symm,
+  apply_fun expand p at key,
+  simp only [expand_bind₁] at key,
+  rw key, clear key,
```

Comment:
I'm thinking this could be rewritten with `suffices key : _`, but the `_` would be too long to make it worth it:
```lean
key : ⇑(bind₁ (λ (i : idx), ⇑(expand p) (⇑(rename (prod.mk i)) (witt_polynomial p ℚ n))))
      (⇑(map (int.cast_ring_hom ℚ)) Φ) =
    ⇑(bind₁ (λ (i : ℕ), ⇑(expand p) (witt_structure_rat p (⇑(map (int.cast_ring_hom ℚ)) Φ) i)))
      (witt_polynomial p ℚ n)
```

Reply:
Yup... that's one of the things that makes this somewhat unpleasant...